src/main/java/com/ibm/watson/modelmesh/payload/AsyncPayloadProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.ibm.watson.modelmesh.payload;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AsyncPayloadProcessor implements PayloadProcessor {
+
+    private static final Logger logger = LoggerFactory.getLogger(AsyncPayloadProcessor.class);
+
+    private final PayloadProcessor delegate;
+
+    private final LinkedBlockingDeque<Payload> payloads;
+
+    private final AtomicInteger dropped;
+
+    public AsyncPayloadProcessor(PayloadProcessor delegate) {
+        this(delegate, 1, TimeUnit.MINUTES, Executors.newScheduledThreadPool(1), 1000);
```

Comment:
I think it may be necessary to set 2 threads here, since one will be permanently occupied and so a second is needed to handle the scheduled tasks.

It is probably important to have the capacity be configurable, since if there are large messages it could quickly blow up available memory with it set to a number this high. Probably the default should also be smaller e.g. 64 or something.

We could also consider having a configurable number of workers. Then we would spawn that many tasks consuming from the queue, so that multiple payloads could be processed in parallel. This might be needed in some cases to "keep up" with a high rate of requests.

Reply:
I've set the default no. of threads to 2 and the default capacity to 64.